airflow/serialization/serialized_objects.py

```diff
@@ -532,6 +540,13 @@ def _serialize_operator_extra_links(cls, operator_extra_links: Iterable[BaseOper
 
         return serialize_operator_extra_links
 
+    @classmethod
+    def _is_inherited_from_dummy_operator(cls, op: BaseOperator) -> bool:
+        """Used to determine if an Operator is inherited from DummyOperator"""
+        if op.task_type == "DummyOperator" or isinstance(op, DummyOperator):
```

Comment:
I did not add check to see if the execute method is empty or not like we discussed @ashb. My thinking for that was as it is a DummyOperator where we list it should do nothing, let's just keep it simple.

WDYT?

Reply:
Should we just simply do `isinstance(op, DummyOperator)`, why check the `task_type`?

Reply:
We have a serialized DAG here, so the isinstance function is not behaving properly. From the isinstance point of view, we have BaseOperator here.

Reply:
One for real dag, one for Serialzed dag I think

Reply:
How about instead of this approach,

Add a method/propery on to BaseOperator:

```
class BaseOperator:
    @property
    def inherits_from_dummy_operator(self):
        getattr(self, '_is_dummy', False)
```

And then

```
class DummyOperator(BaseOperator):
    inherits_from_dummy_operator = True
```

That way this method isn't needed, and at runtime in both cases (real or serialized) we can just look at `op.inherits_from_dummy_operator

Reply:
Yeah if for whatever reason if there is a SerializedOperator, than we need to check `op.task_type`.

Reply:
Using the approach I suggested (if it works), you can just get rid of this method :)

Reply:
Let me give that a try

Reply:
Works like a charm, thanks.

Updated in https://github.com/apache/airflow/pull/12745/commits/0b47a1b9d251a0ac84a366892701585a831fc892